rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -819,10 +819,85 @@ void getFederationElection_whenElectionIsNotInStorage_shouldReturnDefault() {
         // Assert
 
         byte[] expectedElectionEncoded = BridgeSerializationUtils.serializeElection(expectedElection);
+
         assertArrayEquals(expectedElectionEncoded, serializeElection(actualElection));
 
     }
 
+    @Test
+    void getActiveFederationCreationBlockHeight_beforeRSKIP186_storageIsNotAccessedAndReturnsEmpty() {
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
```

Comment:
Rather than activating specific RSKIPs, I think we should start using ActivationConfigsForTest class. 

For pre RSKIP activation we could use the activations object corresponding to the previous network upgrade where the RSKIP was included. And for post, perhaps we could use `all()` so we ensure the test keeps passing with future RSKIPs.

Unless it's something that is expected to work post FINGERROOT but not anymore post ARROWHEAD for example. In that case we could use network upgrade specific activations.

What do you think?

Reply:
We've been doing like that, or at least my self, since the creation of [ActivationConfigsForTest](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/test/java/org/ethereum/config/blockchain/upgrades/ActivationConfigsForTest.java) helper test class. So, I agree with testing using network upgrades instead of specific RSKIPs.

Reply:
I agree.

Reply:
Updated.